DNN Platform/Library/Data/DataProvider.cs

```diff
@@ -187,6 +187,7 @@ private object GetRoleNull(int RoleID)
 
         public abstract IDataReader ExecuteSQL(string sql);
         public abstract IDataReader ExecuteSQL(string sql, int timeoutSec);
+        public abstract T ExecuteSQL<T>(string sql);
```

Comment:
Hm, this is a very open method, easy to shoot oneself in the foot this way.

I'd rather you do one of the following (firs option preferred)

1. use `IDataReader ExecuteSQL(string sql)` and retrieve the result from the reader
2. add a stored procedure ad use the `ExecuteScalar` method

Reply:
Is it worth for such a small feature improvement?

Reply:
You mean the stored procedure @mean2me? It's probably not worth it.

Unless you have a better suggestion I would prefer the former. Instead of adding a new overload please use the `IDataReader` one above